vmm/src/lib.rs

```diff
@@ -1422,6 +1422,15 @@ impl Vmm {
                     VmConfigError::InvalidVcpuCount,
                 ));
             }
+            #[cfg(target_arch = "x86_64")]
```

Comment:
The enforcement of the vcpu count being less than 32 should be valid for all architectures. That's why I am not sure if the MAX_SUPPORTED_CPUS should be defined in `mptable.rs`. I would see this refactored a bit:
- define a maximum supported cpus in mptable with the value of 254. This would be useful if we decide that this crate can be used by other projects and we decide to publish it as an independent crate. This way, other vmms that use this crate will have the maximum number of vcpus correctly enforced for the x86_64 architecture. Besides the change from 32 to 254 no other changes are required in `mptable.rs`.
- define a Firecracker-vmm maximum CPU number. Firecracker shouldn't accept a vcpu number higher than 32. This can be implemented as a custom validator for the `vcpu_count` field of the `VmConfig` structure. That structure is defined in `vmm/src/vmm_config/machine_config.rs`. Since serde does not currently support custom validators, you can check the work-around here: https://github.com/serde-rs/serde/issues/939.

If we are gonna define a maximum allowed value for the vcpus, we should also change the swagger definition of the `MachineConfiguration` resource and add a maximum to vcpu_count. Since this is a customer-impacting change, the `CHANGELOG.md` should also be updated.